lib/analyze.js

```diff
@@ -147,13 +147,20 @@ function gatherMetaData(url, body) {
 
       return foundValue
     }
-
+    const extractCanonical = function() {
+      const canonical = getSimpleValue('link[rel="canonical"]', 'href')
+      if (canonical && canonical.startsWith('/')) {
+        const parsedUrl = Url.parse(url)
+        return `${parsedUrl.protocol}//${parsedUrl.host}${canonical}`
```

Comment:
Will this handle URLs with a port number?

See: https://www.grc.com/port_81.htm
> Sometimes an "off the beaten path" web server will be established to listen not to port 80, but to an alternate related port such as 81, 82, 8080, or 8090.

Reply:
It does.  Host is either hostname or hostname:port